### DISCUSSION TITLE
Unable to Prevent Default in Passive Event Listener Invocation

### DISCUSSION THREAD

**Discussion by realrnvr** [igordanchenko/yet-another-react-lightbox#323] [General]
I’m currently implementing a feature where I need to disable pointer, keyboard, and scroll events when a controller (controlled by an isOpen state) is active.

I attempted to block these events using the subscribeSensors API provided by the library. However, I’m encountering the following error:
**Unable to preventDefault inside passive event listener invocation.**

here is my code:

```
useEffect(() => {
  if (isOpen) {
    return;
  }

  const unsubscribeKeyDown = subscribeSensors("onKeyDown", (event) => {
    event.stopPropagation();
  });

  const unsubscribePointerDown = subscribeSensors("onPointerDown", (event) => {
    event.stopPropagation();
  });

  const unsubscribeWheel = subscribeSensors("onWheel", (event) => {
    event.stopPropagation();
    event.preventDefault(); // Causes the error
  });

  const unsubscribeGestureStart = subscribeSensors("onGestureStart", (event) => {
    event.stopPropagation();
    event.preventDefault();
  });

  return () => {
    unsubscribeKeyDown();
    unsubscribePointerDown();
    unsubscribeWheel();
    unsubscribeGestureStart();
  };
}, [isOpen, subscribeSensors]);
```

Thank you!

**Comment by igordanchenko** [comment]
Can you describe the use case you are trying to implement? 

The lightbox in its default configuration prevents default wheel events over here -

https://github.com/igordanchenko/yet-another-react-lightbox/blob/b91a18ab5b531f5cb2530baba8da1e386f6f3570/src/modules/Controller/usePreventWheelDefaults.ts#L18-L23

**Comment by realrnvr** [comment]
Thank you so much for the prompt response and for pointing me to the specific code snippet! I really appreciate the detailed explanation—it gave me a much better understanding of how the wheelEvent operates under the hood.

After reviewing my previous implementation, I realized that I had missed adding the ! before the isOpen boolean. I’ve now updated the code accordingly.

```
  useEffect(() => {
    if (!isOpen) {
      return;
    }

    const unsubscribeKeyDown = subscribeSensors("onKeyDown", (event) => {
      event.stopPropagation();
    });

    const unsubscribePointerDown = subscribeSensors(
      "onPointerDown",
      (event) => {
        event.stopPropagation();
      }
    );

    const unsubscribeWheel = subscribeSensors("onWheel", (event) => {
      event.stopPropagation();
      event.preventDefault();
    });

    const unsubscribeGestureStart = subscribeSensors(
      "onGestureStart",
      (event) => {
        event.stopPropagation();
        event.preventDefault();
      }
    );

    return () => {
      unsubscribeKeyDown();
      unsubscribePointerDown();
      unsubscribeWheel();
      unsubscribeGestureStart();
    };
  }, [isOpen, subscribeSensors]);
```

To give you a clearer idea of what I’m trying to achieve, I’ve prepared a visual explanation using Excalidraw. Here’s the link: [excalidraw](https://excalidraw.com/#json=FEDztnL2WaP2lKDXvXSA9,tF9lIHkbZVLmcBy9W_K4jw)

I sincerely apologize for any earlier oversights or typos in my explanation. Thanks again for your insights and for helping me better understand the implementation!
